core/eolearn/tests/test_utils/test_parsing.py

```diff
@@ -0,0 +1,142 @@
+from dataclasses import dataclass
+from typing import List, Tuple, Union
+
+import numpy as np
+import pytest
+
+from eolearn.core import EOPatch, FeatureParser, FeatureType
+from eolearn.core.utils.parsing import FeatureRenameSpec, FeatureSpec, FeaturesSpecification
+from eolearn.core.utils.types import EllipsisType
+
+
+@dataclass
+class ParsingTestClass:
+    input: FeaturesSpecification
+    output_get_features: List[FeatureSpec]
+    output_get_renamed_features: List[FeatureRenameSpec]
+    output_get_feature_specifications: List[Tuple[FeatureType, Union[str, EllipsisType]]]
+
+
+TEST_CASES_EMPTY = [
+    ParsingTestClass(
+        input=(), output_get_features=[], output_get_renamed_features=[], output_get_feature_specifications=[]
+    ),
+]
+
+
+TEST_CASES = [
+    ParsingTestClass(
+        input=((FeatureType.DATA, "bands", "new_bands"),),
+        output_get_features=[(FeatureType.DATA, "bands")],
+        output_get_renamed_features=[(FeatureType.DATA, "bands", "new_bands")],
+        output_get_feature_specifications=[(FeatureType.DATA, "bands")],
+    ),
+    ParsingTestClass(
+        input=([(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands", "new_bands")]),
+        output_get_features=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+        output_get_renamed_features=[(FeatureType.DATA, "bands", "bands"), (FeatureType.MASK, "bands", "new_bands")],
+        output_get_feature_specifications=[(FeatureType.DATA, "bands"), (FeatureType.MASK, "bands")],
+    ),
+    ParsingTestClass(
+        input=(
+            {
+                FeatureType.DATA: [("bands", "new_bands"), ("CLP", "new_CLP")],
+                FeatureType.MASK: [("bands", "new_bands"), ("CLP", "new_CLP")],
+            }
+        ),
+        output_get_features=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+        output_get_renamed_features=[
+            (FeatureType.DATA, "bands", "new_bands"),
+            (FeatureType.DATA, "CLP", "new_CLP"),
+            (FeatureType.MASK, "bands", "new_bands"),
+            (FeatureType.MASK, "CLP", "new_CLP"),
+        ],
+        output_get_feature_specifications=[
+            (FeatureType.DATA, "bands"),
+            (FeatureType.DATA, "CLP"),
+            (FeatureType.MASK, "bands"),
+            (FeatureType.MASK, "CLP"),
+        ],
+    ),
+]
+
+
+TEST_CASES_ELLIPSIS = [
```

Comment:
should include a sequence case and `...` and a (ftype, ...) case

Reply:
also a dict case which includes a FeatureType.BBOX or TIMESTAMP

Reply:
done

Reply:
You dont have TIMESTAMP anywhere in the tests, please add it as well since it's different compared to others

Reply:
done